kloppy/infra/serializers/event/wyscout/serializer.py

```diff
@@ -0,0 +1,370 @@
+        qualifiers.append(PassQualifier(PassType.SIMPLE_PASS))
+    elif raw_event["subEventName"] == 86:
+        qualifiers.append(PassQualifier(PassType.SMART_PASS))
+
+    if _has_tag(raw_event, 401):
+        qualifiers.append(BodyPartQualifier(BodyPart.LEFT_FOOT))
+    elif _has_tag(raw_event, 402):
+        qualifiers.append(BodyPartQualifier(BodyPart.RIGHT_FOOT))
+
+    return qualifiers
+
+
+def _parse_pass(raw_event: Dict, next_event: Dict) -> Dict:
+    pass_result = None
+
+    if _has_tag(raw_event, 1801):
+        pass_result = PassResult.COMPLETE
+    elif _has_tag(raw_event, 1802):
+        pass_result = PassResult.INCOMPLETE
+
+    if next_event:
+        if next_event["eventName"] == 6:
+            pass_result = PassResult.OFFSIDE
+        elif next_event["eventName"] == 5 and next_event["subEventName"] == 50:
+            pass_result = PassResult.OUT
+
+    return {
+        "result": pass_result,
+        "qualifiers": _pass_qualifiers(raw_event),
+        "receive_timestamp": None,
+        "receiver_player": None,
+        "receiver_coordinates": Point(**raw_event["positions"][1]),
+    }
+
+
+def _parse_foul(raw_event: Dict) -> Dict:
+    return {"result": None, "qualifiers": None}
+
+
+def _parse_card(raw_event: Dict) -> Dict:
+    card_type = None
+    if _has_tag(raw_event, 1701):
+        card_type = CardType.RED
+    elif _has_tag(raw_event, 1702):
+        card_type = CardType.FIRST_YELLOW
+    elif _has_tag(raw_event, 1703):
+        card_type = CardType.SECOND_YELLOW
+
+    return {"result": None, "qualifiers": None, "card_type": card_type}
+
+
+def _parse_recovery(raw_event: Dict) -> Dict:
+    return {
+        "result": None,
+        "qualifiers": None,
+    }
+
+
+def _parse_ball_out(raw_event: Dict) -> Dict:
+    return {"result": None, "qualifiers": None}
+
+
+def _parse_set_piece(raw_event: Dict, next_event: Dict) -> Dict:
+    qualifiers = []
+
+    result = {}
+
+    if raw_event["subEventName"] in [30, 31, 32, 34, 36]:
+        result = _parse_pass(raw_event, next_event)
+        if raw_event["subEventName"] == 34:
+            qualifiers.append(SetPieceQualifier(SetPieceType.GOAL_KICK))
+        elif raw_event["subEventName"] == 36:
+            qualifiers.append(SetPieceQualifier(SetPieceType.THROW_IN))
+            qualifiers.append(PassQualifier(PassType.HAND_PASS))
+        elif raw_event["subEventName"] in [31, 32]:
+            qualifiers.append(SetPieceQualifier(SetPieceType.FREE_KICK))
+        elif raw_event["subEventName"] == 30:
+            qualifiers.append(SetPieceQualifier(SetPieceType.CORNER_KICK))
+    elif raw_event["subEventName"] in [33, 35]:
+        result = _parse_shot(raw_event, next_event)
+        if raw_event["subEventName"] == 33:
+            qualifiers.append(SetPieceQualifier(SetPieceType.FREE_KICK))
+        elif raw_event["subEventName"] == 35:
+            qualifiers.append(SetPieceQualifier(SetPieceType.PENALTY))
+    else:
+        print(raw_event["subEventName"])
```

Comment:
It should ignore, use logger or raise exception

Reply:
Done